### ISSUE TITLE
Fastai Dataloader `drop_last`

### ISSUE THREAD

**Issue by sachinruk** [fastai/fastai#2416]
The datasets that I work with are generally small. But I have set a batch size of 128 by default since Im training multiple models.

When I create a databunch I noticed that the fastai library sets `drop_last=True` as seen in [this line](https://github.com/fastai/fastai/blob/5e1ebc8456652da8d00200de8de9f11bf83bf8bc/fastai/basic_data.py#L118). This can be quite a problem when my entire dataset is <50 samples. 

Just wondering if this is meant to be the case. If not happy to make a pull request changing it. Doesn't make much sense to me as to why you'd want to drop the last batch anyway. 

Alternatively, I can write some code to change the batch size of the training/ validation sets to be `min(len(data), bs)`.

**Comment by sgugger** [comment]
It is meant to be as having a tiny last batch will cause instability in the batchnorm layers. And if the batch is of size 1, it will cause an error.
You can change this by adding `data.train_dl = data.train_dl.new(drop_last=False)` but for the reasons above, it isn't recommended unless you have a model without batchnorm.
